R/defineModel.R

```diff
@@ -1,18 +1,115 @@
+  lapply(c(minNperItem, boundary, n.plausible, n.iterations, nodes, Msteps),
+         checkmate::assert_numeric, len = 1, lower = 0)
+  lapply(c(converge, deviancechange), checkmate::assert_numeric, len = 1)
+  checkmate::assert_numeric(seed, len = 1, null.ok = TRUE)
+
+  #' assert character via match.arg:
+  # software
   software <- match.arg(arg = software, choices = c("conquest","tam"))
-  if ( software == "conquest" && is.null(conquest.folder) ) {
+  if(software == "conquest" && is.null(conquest.folder)) {
     conquest.folder <- identifyConquestFolder()
   }
-  ismc <- attr(dat, "multicore")
-  dat  <- eatTools::makeDataFrame(dat, name = "dat")
+  # method
+  method <- match.arg(arg = method, choices = c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"))
+
+  #' assert logical:
+  lapply(c(check.for.linking, removeMinNperItem, remove.boundary, remove.no.answers,
+           remove.no.answersHG, remove.missing.items, remove.constant.items,
+           remove.failures, remove.vars.DIF.missing, remove.vars.DIF.constant,
+           verbose),
+         checkmate::assert_logical, len = 1)
+
+  # arguments specific to `conquest`:
+  if(software == "conquest"){
+    # character: dir, conquest.folder, model.statement, export
+    checkmate::assert_character(dir, len = 1)
+    checkmate::assert_character(model.statement, len = 1, null.ok = TRUE)
+    checkmate::assert_character(export)
+    checkmate::assert_directory(conquest.folder)
+    # logical: compute.fit, use.letters, allowAllScoresEverywhere
+    lapply(c(compute.fit, use.letters, allowAllScoresEverywhere),
+           checkmate::assert_logical, len = 1)
+    # character via match.arg: constraints, std.err, distribution, equivalence.table
+    constraints <- match.arg(arg = constraints, choices = c("cases","none","items"))
+    std.err <- match.arg(arg = std.err, choices = c("quick","full","none"))
+    distribution <- match.arg(arg = distribution, choices = c("normal","discrete"))
+    equivalence.table <- match.arg(arg = equivalence.table, choices = c("wle","mle","NULL"))
+    # numeric: p.nodes, f.nodes
+    lapply(c(p.nodes, f.nodes), checkmate::assert_numeric, len = 1)
+  }
+
+  # arguments specific to `tam`:
+  if(software == "tam"){
+    # character: dir
+    checkmate::assert_character(dir, len = 1, null.ok = TRUE)
+    # character via match.arg: pvMethod, constraints
+    pvMethod <- match.arg(arg = pvMethod, choices = c("regular", "bayesian"))
+
+    constraints <- match.arg(arg = constraints, choices = c("cases", "none", "items"))
+    if(constraints == "none"){
+      stop("tbd: You can't use constraints = 'none' when using 'tam'.")
+    }
+    # logical: fitTamMmlForBayesian, progress
+    lapply(c(fitTamMmlForBayesian, progress), checkmate::assert_logical, len = 1)
+    # named data frames: guessMat, est.slopegroups, fixSlopeMat
+    lapply(c(guessMat, est.slopegroups, fixSlopeMat), checkmate::assert_data_frame, col.names = "named", ncols = 2)
```

Comment:
I think this leads to an error: 
![Unbenannt](https://github.com/user-attachments/assets/c6b4f0bc-ea71-49ae-9cfe-387940cddc96)
Moreover, `guessMat`, `est.slopegroups`, and `fixSlopeMat` are allowed to be `NULL`.